blockchain/ethereum/contracts/compile and deploy contract preset/f4f89c90-1285-4803-b3c8-530254056164.json

```diff
@@ -0,0 +1,204 @@
+{
+  "name": "blockchain/ethereum/contracts/compile and deploy contract preset",
+  "interface": {
+    "inputs": {
+      "b53990d6-c237-4d59-9dbf-b2fa712181a7": {
+        "type": "{\n    \"contract template\":  string,\n    \"name to replace\":  string,\n    \"compile settings\": {\n        \"includeRoots\": string[],\n        \"solidityVersion\": string,\n        \"compileOptions\": {\n            optional \"evmVersion\": string,\n            \"optimizer\": {\n                \"enabled\": boolean,\n                \"runs\": number\n            }\n        }\n    },\n\"deploy settings\": {\n        \"gas\": number,\n        optional \"gasPrice\": number\n\n    }\n}",
+        "name": "contract description",
+        "description": "Receives a contract preset description."
+      },
+      "741d010c-14a6-4dd1-b528-0d5386187bb7": {
+        "type": "string",
+        "name": "contract name",
+        "description": "Receives the desired contract name to be inserted into the contract preset template."
+      },
+      "67510810-61e0-4c9d-b06d-10fbe7c4a740": {
+        "type": "string[]",
+        "name": "contract params",
+        "description": "An array of arguments to be passed to the deployed contract's constructor when deployed."
+      },
+      "11c0a845-1293-4670-9390-0a0586122a6f": {
+        "type": "typeof `connect credentials` of `deploy contract`",
+        "name": "credentials"
+      }
+    },
+    "outputs": {
+      "2acecfad-46ec-48f8-80ea-332931523eef": {
+        "type": "typeof `events` of `deploy contract`",
+        "name": "events",
+        "description": "[Inherited from port `events` of `contract deployer`] \nMultiplexed events. Useful for monitoring progress.\nUse Demultiplexer to separate them."
```

Comment:
On resusable prototye I would avoid the automatically generated [Inherited ...] text.